lib/puma/dsl.rb

```diff
@@ -588,6 +592,17 @@ def after_worker_fork(&block)
       @options[:after_worker_fork] << block
     end
 
+    # Code to run after puma is booted in a single mode
+    #
+    # @note Single mode only.
+    # @example
+    #   on_booted do
+    #     puts 'After booting...'
+    #   end
+    def on_booted(&block)
+      @options[:events].on_booted(&block)
+    end
```

Comment:
I'm not sure *exactly* why the tests are failing, but (for what it's worth) changing this method makes the tests pass:
```suggestion
    def on_booted(&block)
      @options[:on_booted] ||= []
      @options[:on_booted] << block
    end
```

Reply:
Thanks for the feedback! When you scroll through the history of this PR you'll actually see that was my initial implementation. I've the a suggestion that it might be better to use Puma event system instead